Add Credential Vertical To Share URL (#10)
dimmyvi/secure-credential-transfer: draft-secure-credential-transfer.md

- Secret - a symmetric encryption key shared by a pair of Sender and Receiver devices, used to encrypt Provisioning Information stored on the Relay server. Secret stays the same for the entire credential transfer flow (one Secret per complete transfer). Provisioning Information stored on Relay server is always encrypted using the Secret. In Stateful flow all information exchanged by Sender and Receiver devices through Relay server is encrypted with the same Secret. Thus, effectively, Secret has a one-to-one relation with the mailbox.

> - Credential Vertical - The broad industry vertical that the credential belongs to.

Reviewer: I suggest to add examples here

~~~
{: #stateful-flow-image title="Sample stateful workflow"}

> ## Credential Vertical in share url

Author: I wasn't entirely sure on the best place for this section. What does everyone think?

Reviewer: It seems we have definitions of parts of the URL scattered across the spec, so I would say adding a dedicated section to define the URL and it's parts, including if they are required\optional, would be better.

Author: I noticed that too, let me combine the various URL sections.

Author: I added a more general share url section, but I wasn't how much detail to go into for each field. Do you think it would be worth recapping a description for each field in this new section?

> When a user interacts with a share url on a Receiver device it can be helpful to know what Credential Vertical this share is for. This is particularly important if the Receiver device has multiple apps that can handle a share url. For example, a Receiver device might want to handle a general access share in their wallet app, but handle car key shares in a special car app.

Reviewer 2: ```suggestion
When a user interacts with a share URL on a Receiver device it can be helpful to know what Credential Vertical this share is for. This is particularly important if the Receiver device has multiple applications that can handle a share URL. For example, a Receiver device might want to handle a general access share in their wallet app, but handle car key shares in a specific car application.
```

> To allow for this kind of share url universal link routing the sender can include the Credential Vertical in the share url as a query parameter. The Credential Vertical can't be included in the encrypted payload because the Receiver device might need to open the right app before retrieving the secure payload. The Credential Vertical query parameter uses the "v" key and supports the below types. If no Credential Vertical is provided it will be assumed that this is a general access share url.

Reviewer 2: ```suggestion
To properly route a share URL, the sender can include the Credential Vertical in the share URL as a query parameter. The Credential Vertical can't be included in the encrypted payload because the Receiver device might need to open the right application before retrieving the secure payload. The Credential Vertical query parameter uses the "v" key and supports the below types. If no Credential Vertical is provided it will be assumed that this is a general access share URL.
```

| Vertical       | Value |
| --------       | ----- |
| General Access | a or *None*
| Home Key       | h
| Car Key        | c

Share url structure:
```

> http://relayserver.com/v1/m/{mailboxIdentifier}?v={vertical}

Reviewer: ```suggestion
http://relayserver.com/v1/m/{mailboxIdentifier}?v={vertical}#{Secret}
```

```

Example car key share url:
```

> http://relayserver.com/v1/m/2bba630e-519b-11ec-bf63-0242ac130002?v=c

Reviewer: maybe also add the secret so the example has all the possible parts of the URL?

```

> The Credential Vertical query parameter should be added to the share URL by the Sender device when constructing the full share URL that is going to be sent to the Receiver device.

Reviewer 3: This sentence sounds like the credential Vertical query parameter is mandatory but the previous section mentions that "If no Credential Vertical is provided it will be assumed that this is a general access share URL."

Should we just make it mandatory consistently across the sections? I don't see why the sender device do not why to specify this

Author: Good point. I reworded "should" to "can", and then I think that with the table above should make it clear that it is optional.

> The Credential Vertical query parameter should be added to the share url by the Sender device when constructing the full share url that is going to be sent to the Receiver device.

Reviewer 2: ```suggestion
The Credential Vertical query parameter should be added to the share URL by the Sender device when constructing the full share URL that is going to be sent to the Receiver device.
```
